packages/math-input/src/components/input/key-handlers/handle-backspace.ts

```diff
@@ -0,0 +1,275 @@
+    // has a subscript as is the case in log_n then move the cursor into
+    // the subscript, e.g. \log_{5}(|x+1) => \log_{5|}(x+1)
+
+    if (grandparent[MQ.L].sub) {
+        // if there is a subscript
+        if (grandparent[MQ.L].sub.jQ.text()) {
+            // and it contains text
+            // move the cursor to the right end of the subscript
+            cursor.insAtRightEnd(grandparent[MQ.L].sub);
+            return;
+        }
+    }
+
+    // Determine if the parens are empty before we modify the
+    // cursor's position.
+    const isEmpty = isInsideEmptyNode(cursor);
+
+    // Insert the cursor to the left of the command if there is one
+    // or before the '\\left(` if there isn't
+    const command = maybeFindCommandBeforeParens(grandparent);
+
+    cursor.insLeftOf((command && command.startNode) || grandparent);
+    cursor.startSelection();
+    cursor.insRightOf(grandparent);
+    cursor.select();
+    cursor.endSelection();
+
+    // Delete the selection, but only if the parens were empty to
+    // begin with.
+    if (isEmpty) {
+        mathField.keystroke("Backspace");
+    }
+}
+
+function handleBackspaceAfterLigaturedSymbol(mathField: MathFieldInterface) {
+    mathField.keystroke("Backspace");
+    mathField.keystroke("Backspace");
+}
+
+/**
+ * Selects and deletes part of the expression based on the cursor location.
+ * See inline comments for precise behavior of different cases.
+ */
+function handleBackspace(mathField: MathFieldInterface) {
+    const cursor = getCursor(mathField);
+    if (!cursor.selection) {
+        const parent = cursor.parent;
+        const grandparent = parent.parent;
+        const leftNode = cursor[MQ.L];
+
+        if (isFraction(leftNode)) {
+            selectNode(leftNode, cursor);
+        } else if (isSquareRoot(leftNode)) {
+            selectNode(leftNode, cursor);
+        } else if (isNthRoot(leftNode)) {
+            selectNode(leftNode, cursor);
+        } else if (isNthRootIndex(parent)) {
+            handleBackspaceInRootIndex(mathField, cursor);
+        } else if (leftNode.ctrlSeq === "\\left(") {
+            handleBackspaceOutsideParens(cursor);
+        } else if (grandparent.ctrlSeq === "\\left(") {
+            handleBackspaceInsideParens(mathField, cursor);
+        } else if (isInsideLogIndex(cursor)) {
+            handleBackspaceInLogIndex(mathField, cursor);
+        } else if (
+            leftNode.ctrlSeq === "\\ge " ||
+            leftNode.ctrlSeq === "\\le "
+        ) {
+            handleBackspaceAfterLigaturedSymbol(mathField);
+        } else if (
+            isNthRoot(grandparent) &&
+            leftNode === MathFieldActionType.MQ_END
+        ) {
+            handleBackspaceInNthRoot(mathField, cursor);
+        } else {
+            mathField.keystroke("Backspace");
+        }
+    } else {
+        mathField.keystroke("Backspace");
+    }
+}
+
+export default handleBackspace;
```

Comment:
The nice thing about this refactor is now all the other functions in this file are private to the file. We just have one export and rest easy that the other functions are only used here.